data/publiccloud/log_instance.sh

```diff
@@ -0,0 +1,182 @@
+#!/bin/bash
+
+COMMAND=$1
+PROVIDER=$2
+INSTANCE_ID=$3
+HOST=$4
+ZONE=$5
+OUTPUT_DIR=/tmp/log_instance/"$INSTANCE_ID"
+LOCK=${OUTPUT_DIR}/.lock
```

Comment:
can you elaborate need of the lock ?

Reply:
It's a script which creates background processes and store status information on the fs. For this information I introduced the lock. So parallel execution of it, doesn't show unexpected behavior.

Reply:
don't want to dive too deep into endless `over-engineering` discussion . But if you have VM which **you** are starting and you have a script which **you** are starting there how you can end up with two instances of this script running at the same time ?

Reply:
Sure it is not very likely but technical possible during debugging or using multiple consoles in parallel.

Reply:
what you mean "using several consoles" ? you mean manually ? but how it is related  to automated test ?

Reply:
I mean several consoles in case of `root-console`, `user-console` and so on.

Reply:
In general, this is currently used only in `run_ltp.pm` but we need to add it to others as well. And there we will get potential more possibilities.

Reply:
> In general, this is currently used only in `run_ltp.pm` but we need to add it to others as well. And there we will get potential more possibilities.

this is what I am talking about , in **current** patch only place where this script is called is run_ltp.pm . Each call of this script is done with unique combination of provider / ip address / instance id which is unique with **current** test design. Introducing heavy locking mechanism just because that **there is a chance** that we **probably** add some code which **probably** may introduce executing several instances of this script running on same VM I would call this too much

Reply:
I was also wondering why you implemented this lock... this script is to be executed inside the cloud helper, which is a different VM for each test, right? So then, this situation you are trying to avoid happens only when doing manual debugging, and never in automated tests. Is it right or am I missing something?

Reply:
Ok I see, you don't like the locking :) and I agree we can avoid the logging of the counter variable and pid file.
But I think I will keep the locking (or some other sync) for the polling bg process. So I kill it only on a sleep.

Reply:
done